internal/controlplane/handlers_githubwebhooks.go

```diff
@@ -266,64 +259,10 @@ func (s *Server) registerWebhookForRepository(
 		return regResult, nil
 	}
 
-	urlUUID := uuid.New().String()
-
-	webhookUrl := fmt.Sprintf("%s/%s", url, urlUUID)
-	parsedOriginalURL, err := urlparser.Parse(webhookUrl)
+	hookUUID, githubHook, err := s.webhookManager.CreateWebhook(ctx, client, repo.Owner, repo.Name)
```

Comment:
I agree that we should move webhook creation to a dedicated interface, but I'm not sure I'm too keen on this way of doing it. I was thinking more along the lines of having a dedicated "webhook-setter" provider type or trait which would be instantiated by the provider builder, and we'd then shove the github-related logic in there. The idea is that there are other systems (e.g. DockerHub) which support webhook installation, so this is something that belongs more on the provider side as opposed to logic that should go in the server.

Reply:
I think this will be a good idea once we build another provider which needs to manage webhooks. At that point in time, it will become much more clear what a generic webhook manager interface may look like. Until then, I think we should have a GitHub-specific interface, and we can refactor the logic to work with the generic interface once we build it.

Reply:
I think in the future, when we

```
// TODO: the remaining logic should be refactored into a repository
// registration interface
```

The webhookManager would be part of the repository registration interface, not part of `Server`, correct?  If so, this seems like an okay intermediate step.

Reply:
@evankanderson Yes, and I even have a draft PR sitting around locally where I have made this change already.